pkg/services/store/entity/sqlstash/broadcaster.go

```diff
@@ -0,0 +1,174 @@
+package sqlstash
+
+import (
+	"context"
+	"fmt"
+	"sync"
+)
+
+type ConnectFunc[T any] func() (chan T, error)
+
+type Broadcaster[T any] struct {
```

Comment:
NIT – I recognize this code from elsewhere – it's not the most clean one. I think it's OK to keep it for now but we may want to consider something like https://pkg.go.dev/github.com/dustin/go-broadcast instead.

Reply:
channel based reg/unreg to avoid the need for mutex is neat, I'll look into that approach.